WordPressKit/PlanServiceV3Remote.swift

```diff
@@ -0,0 +1,69 @@
+import Foundation
+import WordPressShared
+import CocoaLumberjack
+
+@objc public class PlanServiceV3Remote: ServiceRemoteWordPressComREST {
```

Comment:
I'm still not sure having a separate class is necessary. `PlanServiceV3Remote` and `PlanServiceRemote` do the same thing. The only difference is how they parse the response and what is returned. Would it not be possible to add an api version parameter to `getPlansForSite`  and decide how to proceed based on that parameter?

Reply:
I can sure remove the PlanServiceV3Remote class and add a new method to PlanServiceRemote. Actually at the beginning that was what I did that but I had to add a suffix like getPlansForSite_version1_3 I choosed adding the suffix to the class instead but I can revert.

But, on the other hand merging them both into the old getPlansForSite is very problematic. Because then I need to make them return the same response in the success completion, this means merging all parameters of v2 and v3 in the same response model, and there is a huge difference between them:

old one:
```
 "product_slug": "free_plan",
  "description": "",
  "cost": 0,
  "bill_period": -1,
  "product_type": "bundle",
  "available": "yes",
  "multi": 0,
  "bd_slug": "wp-free-plan",
  "bd_variation_slug": "wp-bundles",
  "outer_slug": null,
  "extra": null,
  "capability": "manage_options",
  "product_name_short": "Free",
  "icon": "https:\/\/s0.wordpress.com\/i\/store\/plan-free.png",
  "icon_active": "https:\/\/s0.wordpress.com\/i\/store\/plan-free-active.png",
  "bill_period_label": "for life",
  "price": "$0",
  "formatted_price": "$0",
  "raw_price": 0,
  "tagline": "Just get started",
  "currency_code": "USD",
  "features_highlight": [
                         {
                         "items": [
                                   "free-blog",
                                   "space",
                                   "support"
                                   ]
                         }
                         ],
  "formatted_original_price": "$0",
  "raw_discount": 0,
  "formatted_discount": "$0",
  "current_plan": true
  }
```
new one:
```
  "1008" : {
    "current_plan" : true,
    "discount_reason" : null,
    "currency_code" : "USD",
    "user_facing_expiry" : "2019-07-07",
    "auto_renew" : true,
    "product_name" : "WordPress.com Business",
    "formatted_original_price" : "$0",
    "formatted_price" : "$300",
    "raw_discount" : 0,
    "product_slug" : "business-bundle",
    "has_domain_credit" : true,
    "raw_price" : 300,
    "free_trial" : false,
    "id" : "10872031",
    "is_domain_upgrade" : null,
    "subscribed_date" : "2018-07-10 19:41:28",
    "auto_renew_date" : "2019-06-10",
    "expiry" : "2019-07-10",
    "user_is_owner" : true,
    "interval" : 365,
    "formatted_discount" : "$0"
  }
```